### PR TITLE
Add MV LSTM model, Combination 2D, Kmax Pooling

### PR DESCRIPTION
**Pull request by fareise**
Add MV LSTM model, Combination 2D, Kmax Pooling

### CODE REVIEW

File: block_zoo/PoolingKmax2D.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            self.output_dim.append((self.input_dims[0][2] + 2 * self.padding[1] - self.window_size[1]) // self.stride[1] + 1)
+        else:
+            self.output_dim.append(-1)
+        # print("pool",self.output_dim)
+        self.input_channel_num = self.input_dims[0][-1]
+
+        self.output_dim.append(self.input_dims[0][-1])
+        '''
+        self.output_dim = [self.input_dims[0][0], -self.input_dims[0][1] * self.k]   #?怎么设定维度，input_dims都是-1
+        #print("&^&^&^")
```
Comment (adolphk-yk) [inline block_zoo/PoolingKmax2D.py]: suggest to remove some debug code(print messages)

File: block_zoo/op/Combination2D.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        Args:
+            args (list): [string, string_len, string2, string2_len, ...]
+                e.g. string (Variable): [batch_size, dim], string_len (ndarray): [batch_size]
+
+        Returns:
+            Variable: [batch_size, output_dim], None
+
+        """
+        '''
+        输入两个string都是[batch_size, seq_len, hidden_size]的
```
Comment (adolphk-yk) [inline block_zoo/op/Combination2D.py]: annotation changes to English

File: block_zoo/PoolingKmax2D.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+        Args:
+            string (Tensor): tensor with shape: [batch_size, length, width, feature_dim]
+            string_len (Tensor): [batch_size], default is None.
+
+        Returns:
+            Tensor: Pooling result of string
+
+        """
+        #print("&&&&&&")
```
Comment (adolphk-yk) [inline block_zoo/PoolingKmax2D.py]: suggest to delete debug messages(print code)

File: block_zoo/op/Combination2D.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        self.layer_conf = layer_conf
+        #print("**************")
+        #print(layer_conf.input_dims)   #[[-1, -1, 256], [-1, -1, 256]]
+        #self.linear_bi = nn.Linear(layer_conf.input_dims[0][2], layer_conf.input_dims[0][2])
+        #self.linear_ten1 = nn.Linear(layer_conf.input_dims[0][2], layer_conf.input_dims[0][2])
+        #self.linear_ten2 = nn.Linear(layer_conf.input_dims[0][1], layer_conf.input_dims[0][2])
+
+        logging.warning("The length Combination layer returns is the length of first input")
+
+    def forward(self, *args):
```
Comment (boshining) [inline block_zoo/op/Combination2D.py]: *args can be x, x_len, y_, y_len

File: block_zoo/op/Combination2D.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        self.operations = ["origin", "difference", "dot_multiply", "cosine", "bilinear", "tensor"]
+
+    @DocInherit
+    def declare(self):
+        self.num_of_inputs = -1
+        self.input_ranks = [-1]
+
+    @DocInherit
+    def inference(self):
+        #说明上一层的dim就有问题！
```
Comment (boshining) [inline block_zoo/op/Combination2D.py]: Annotation in English is better

File: block_zoo/PoolingKmax2D.py [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+            self.pool = nn.MaxPool2d(kernel_size=layer_conf.window_size,stride=layer_conf.stride,padding=layer_conf.padding)
+        elif layer_conf.pool_type == "mean":
+            self.pool = nn.AvgPool2d(kernel_size=layer_conf.window_size,stride=layer_conf.stride,padding=layer_conf.padding)
+        self.k = layer_conf.k
+
+    def forward(self, string, string_len=None):
+        """ process inputs
+
+        Args:
+            string (Tensor): tensor with shape: [batch_size, length, width, feature_dim]
```
Comment (adolphk-yk) [inline block_zoo/PoolingKmax2D.py]: what about 3_dim input [batch_size, seq_length, dim]

File: block_zoo/PoolingKmax2D.py [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+class PoolingKmax2D(BaseLayer):
+    """ Pooling layer
+
+    Args:
+        layer_conf (PoolingConf): configuration of a layer
+    """
+    def __init__(self, layer_conf):
+        super(PoolingKmax2D, self).__init__(layer_conf)
+        self.pool = None
```
Comment (adolphk-yk) [inline block_zoo/PoolingKmax2D.py]: The pool attribute is not used at forward below.

File: block_zoo/PoolingKmax2D.py [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+        #print(string.size())   #[8, 11, 89]
+
+
+        #string = string.permute([0,3,1,2]).contiguous()
+
+        #string = self.pool(string)
+
+        #string = string.permute([0,2,3,1]).contiguous()
+        string = string.view(string.size()[0], string.size()[1], -1)
+        index = string.topk(self.k, dim=-1)[1].sort(dim=-1)[0]
```
Comment (adolphk-yk) [inline block_zoo/PoolingKmax2D.py]: Topk can realize max k pooling, what about mean k pooling(the pool_type above can assign 'max' and 'mean')?

File: block_zoo/PoolingKmax2D.py [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+        self.pool_type = 'max'  # Supported: ['max']
+        self.k = 50
+        
+    @DocInherit
+    def declare(self):
+        self.num_of_inputs = 1
+        self.input_ranks = [4]
+
+    
+    def check_size(self, value, attr):
```
Comment (adolphk-yk) [inline block_zoo/PoolingKmax2D.py]: delete unused function

File: block_zoo/op/Combination2D.py [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+
+        logging.warning("The length Combination layer returns is the length of first input")
+
+    def forward(self, *args):
+        """ process inputs
+        Args:
+            args (list): [string, string_len, string2, string2_len, ...]
+                e.g. string (Variable): [batch_size, dim], string_len (ndarray): [batch_size]
+        Returns:
+            Variable: [batch_size, output_dim], None
```
Comment (boshining) [inline block_zoo/op/Combination2D.py]: this dimension is different from output_dim?
